src/mixins/animation.mixin.js

```diff
@@ -174,6 +174,8 @@ fabric.util.object.extend(fabric.Object.prototype, /** @lends fabric.Object.prot
       propPair = property.split('.');
     }
 
+    var propIsText = _this.colorProperties.includes(property) || (propPair && _this.colorProperties.includes(propPair[propPair.length - 1]));
```

Comment:
we still support full es5 browsers, and includes is not part of it, you have to use indexOf.

Also propPair is supported only as a 2 word key, like `shadow.color` please use propPair[1] instead of `length - 1`

Reply:
sure thing, though actually this is also another issue i've faed when updating points-based shapes (e.g. Polyline) as it requires at least a three-level props tree. Would it be something you'd consider useful to allow for more than just two-level? i made a fix for my project that works well, so i can always make another PR